### DISCUSSION TITLE
wmts service

### DISCUSSION THREAD

**Discussion by baihaohan** [pelicanmapping/osgearth#2639] [Support Q&A] [answered]
I failed to add the wmts service using xyzimagelayer, and there was an error [osgearth]*[geopoint] called toworld() on an invalid point.
Below is my code and the xml file of the service and the image of the error and service effect.

**Comment by gwaldron** [comment]
Your image did not upload. Please paste your earth file or a way to reproduce your issue.

**Reply by baihaohan** [reply]
Ok, I've re-uploaded

**Comment by gwaldron** [comment, marked as answer]
WMTS requires that you use the `GDALImageLayer` and set the `connection` to a WMTS endpoint, like so:
```c++
auto layer = new GDALImageLayer();
layer->setConnection("WMTS:https://services.arcgisonline.com/arcgis/rest/services/World_Imagery/MapServer/WMTS/1.0.0/WMTSCapabilities.xml,layer=World_Imagery");
```
(Notice the `WMTS:` prefix on the URL.)
